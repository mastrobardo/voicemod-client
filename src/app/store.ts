import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import { soundsApi } from '../features/MainView/main-view.service';
import soundDetailsSliceReducer from '../features/SoundDetails/sound-details.slice';
import logger from 'redux-logger';

export const store = configureStore({
  reducer: {
    [soundsApi.reducerPath]: soundsApi.reducer,
    soundsDetails: soundDetailsSliceReducer
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat([soundsApi.middleware, logger]),
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
