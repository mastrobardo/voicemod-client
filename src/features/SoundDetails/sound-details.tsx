import { Modal, Box, Typography, Button, Card, CardActions, CardContent, CardMedia } from "@mui/material";
import useSound from "use-sound";
import { useAppDispatch } from "../../app/hooks";
import { soundsApi } from "../MainView/main-view.service";
import { unsetModalId } from "./sound-details.slice";

type SoundDetailsModal = {
    id: string;
    open: boolean;
}

const style = {
    position: 'absolute' as 'absolute',
    top: '50%',
    left: '50%',
    transform: 'translate(-50%, -50%)',
    width: 400,
    bgcolor: 'background.paper',
    border: '2px solid #000',
    boxShadow: 24,
    p: 4,
  };

export const SoundDetails = ({ id, open } :SoundDetailsModal) => {
    const { data } = soundsApi.endpoints.getSound.useQuery(id, {
            skip: Boolean(id) === false
        });
    const dispatch = useAppDispatch();
    const [play] = useSound('sounds/sound.mp3');
    const onCloseHandler = () => dispatch(unsetModalId());
    const onClickSoundHandler = () => {
        if(data) {
            dispatch(soundsApi.endpoints.updateSound.initiate(data._id));
            play();
        }
    }
    return (
            data ? 
            <Modal
            open={open}
            onClose={() => onCloseHandler()}
            aria-labelledby="modal-modal-title"
            aria-describedby="modal-modal-description"
            >
            <Box sx={style} >
            <Card data-id={data._id}>
                <CardMedia
                    component="img"
                    height="140"
                    image={data.icon}
                    alt={data.name}
                />
                <CardContent>
                    <Typography gutterBottom variant="h2" component="div">
                    {data.name}
                    </Typography>
                    <Typography gutterBottom variant="h5" component="div">
                    Playbacks: {data.playbacks}
                    </Typography>
                    <Typography gutterBottom variant="h5" component="div">
                    Price: {data.price}
                    </Typography>
                </CardContent>
                <CardActions>
                    <Button size="small" onClick={() => onClickSoundHandler()}> listen Sound</Button>
                </CardActions>
                </Card>
            </Box>
            </Modal> : <></>
            )
}