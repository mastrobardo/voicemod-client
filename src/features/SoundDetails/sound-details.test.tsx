import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { store } from '../../app/store';

import { SoundDetails } from './sound-details';
describe('<SoundDetails />', () => {
    const initialState = {}    

    it('renders correctly when there are no items', () => {
        const tree = renderer.create(<Provider store={store}><SoundDetails /></Provider>).toJSON();
        expect(tree).toMatchSnapshot();
      });
})
