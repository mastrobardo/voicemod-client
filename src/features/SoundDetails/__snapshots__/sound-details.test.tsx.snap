// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SoundDetails /> renders correctly when there are no items 1`] = `null`;
